### PR TITLE
fix: Currency rate by date always returns last known rate (#12)

### PR DESCRIPTION
**Pull request by qkoshelev**

### CODE REVIEW

File: api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt [hunk 1]
```diff
@@ -86,6 +87,42 @@ internal class CurrencyControllerFt(
         assertEquals(currencyRate?.rate, rateValue)
     }
 
+    @Test
+    fun `get actual polygon with date filter`() = runBlocking {
```
Comment (yakischik) [inline api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt:91]: Mention of polygon is redundant here

File: api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt [hunk 2]
```diff
@@ -86,6 +87,42 @@ internal class CurrencyControllerFt(
...
+        val rateValue = saveRate("222.22", dateBefore2)
+        val rateValueLatest = saveRate("333.33", dateAfter)
+
+        val currencyRateBeforeDate = getRateForDate(dateBefore2.plusSeconds(1))
+        assertEquals(currencyRateBeforeDate?.rate, rateValue)
+
+        val currencyRateLatest = getRateForDate(dateAfter.plusSeconds(1))
+        assertEquals(currencyRateLatest?.rate, rateValueLatest)
+
+        val currencyRateLatestBeforeDate = getRateForDate(dateAfter.minusSeconds(1))
```
Comment (yakischik) [inline api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt:107]: Here is minus seconds, but in private method - plus. Not sure this is correct

File: api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt [hunk 3]
```diff
@@ -86,6 +87,42 @@ internal class CurrencyControllerFt(
         assertEquals(currencyRate?.rate, rateValue)
     }
 
+    @Test
+    fun `get actual polygon with date filter`() = runBlocking {
+
+        val dateBefore1 = Instant.now().minusSeconds(120)
```
Comment (yakischik) [inline api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt:93]: Use nowMillis() from, regular Instant constructor can include nanos, but mongo trim it, such difference can lead test to fail

**Review by yakischik — CHANGES_REQUESTED** [review]

File: api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt (hunk 2 above)
Comment (qkoshelev) [reply api/src/test/kotlin/com/rarible/protocol/currency/api/controller/CurrencyControllerFt.kt:107]: It was a refactoring mistake. Fixed.

**Pull request merged**